### PR TITLE
Blog update : 2015-06-09

### PR DESCRIPTION
**Pull request by amitjamadagni**
Blog updates for the first two weeks.

### CODE REVIEW

File: news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  b. Common `QuPropagatorState` which has the current state, the current time and the current time state (the next index from time step list)
+  c. `QuPropagatorMethod` parent of all the numerical methods.
+  d. Docile integration
+  e. Tests
+
+The idea of `QuPropagator` is to unify the creation of parameters involved in solving the differential equation. Its creation also involves passing of the numerical method used to solve the problem. So for example:
+
+```
+Hamiltonian = sigmax
+Initial state(qv) = normalize!(QuArray([0.5+0.1im, 0.2+0.2im]))
```
Comment (acroy) [inline news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md]: I would make this Julia syntax, i.e.
```
# initial state
qv = ...
# time steps
t = ...
```

Note that you don't need the parentheses `[...]` for the latter, a Range will also work and is (probably) more efficient.

File: news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+For the above parameters, we construct the `QuPropagator` by :
+
+```julia
+QuPropagator(sigmax, qv, t, QuEuler())
+```
+
+The last parameter being the subtype of `QuPropagatorMethod` which is the numerical method to be used to solve the equation (in this case
+we are using the Euler Method).
+
+Similarly, we could use Crank Nicolson Method by passing `QuCN()` as the last parameter.
```
Comment (acroy) [inline news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md]: Now it is called `QuCrankNicolson`.

File: news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+Presently, we have the following numerical methods :
+<pre>
+QuPropagatorMethod
+|-- QuEuler
+|-- QuCN
```
Comment (acroy) [inline news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md]: `QuCN` -> `QuCrankNicolson`

**Comment by acroy** [comment]
I think it would be nice to give an example of how we think propagators should be used. For instance,
```
psi0 = QuArray([1.,0.])

ep = QuPropagator(sigmax, psi0, 0.:0.1:2pi, QuEuler())
for (t, psi) in ep
    println("$t $(abs2(psi[1])) $(abs2(psi[2]))")
end

cnp = QuPropagator(sigmax, psi0, 0.:0.1:2pi, QuCrankNicolson())
for (t, psi) in cnp
    println("$t $(abs2(psi[1])) $(abs2(psi[2]))")
end

kp = QuPropagator(sigmax, psi0, 0.:0.1:2pi, QuKrylov())
for (t, psi) in kp
    println("$t $(abs2(psi[1])) $(abs2(psi[2]))")
end
```
This is not a very impressive example, but IMO it shows where we are going ...

**Comment by amitjamadagni** [comment]
@acroy a review would be helpful.

File: news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+```
+Hamiltonian = sigmax
+Initial state(qv) = normalize!(QuArray([0.5+0.1im, 0.2+0.2im]))
+Time steps(t) = [0.:0.1:2*pi]
+```
+
+For the above parameters, we construct the `QuPropagator` by :
+
+```julia
+QuPropagator(sigmax, qv, t, QuEuler())
```
Comment (acroy) [inline news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md]: Maybe, `sigmax`->`Hamiltonian`?

**Comment by acroy** [comment]
Have you already explained how to use QuDynamics in the last blog post? Maybe you can briefly mention that one has to clone QuBase and QuDynamics (basically copy from README.md).

File: news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md (hunk 4 above)
Comment (amitjamadagni) [inline news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md]: I get an error using this 
```julia
ERROR: `eye` has no method matching eye(::Function)
```

Comment (acroy) [inline news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md]: I guess you had already defined `Hamiltonian` to be something else then? Try `Hamiltonian = sigmax` and then `QuPropagator(Hamiltonian, qv, t, QuEuler())`. If this doesn't work, try to start in a new REPL session.

Comment (amitjamadagni) [inline news/_posts/2015-06-09-GSoC2015:Week_1_and_2_Updates.md]: Ah my bad, sorry I was really thinking off on this. Done with the edits !

**Pull request merged**